client/dashboard/domains/domain-dns/email-setup-form.tsx

```diff
@@ -0,0 +1,147 @@
+import { useMutation } from '@tanstack/react-query';
+import {
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+	Button,
+} from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { DataForm, Field, isItemValid } from '@wordpress/dataviews';
+import { __, sprintf } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { useState } from 'react';
+import { domainDnsApplyTemplateMutation } from '../../app/queries/domain-dns-records';
+import { domainRoute } from '../../app/routes/domain-routes';
+
+export type EmailSetupFormData = {
+	record: string;
+};
+
+const form = {
+	type: 'regular' as const,
+	fields: [ 'record' ],
+};
+
+const defaultFormData: EmailSetupFormData = {
+	record: '',
+};
+
+interface EmailSetupFormProps {
+	description: string;
+	label: string;
+	modifyVariables?: ( variables: { token: string; domain: string; mxdata?: string } ) => {
```

Comment:
[super nit] a more common term for this use case is `transform` instead of `modify` 😄